pkg/policy/consumer.go

```diff
@@ -216,40 +200,43 @@ func (c *Consumable) removeFromMaps(id NumericIdentity) {
 		})
 
 		scopedLog.Debug("Updating policy BPF map: denying Identity")
-		if err := m.DeleteConsumer(id.Uint32()); err != nil {
+		if err := m.DeleteIdentity(id.Uint32()); err != nil {
 			scopedLog.WithError(err).Warn("Update of policy map failed")
 		}
 	}
 }
 
-// AllowConsumerLocked adds the given consumer ID to the Consumable's
-// consumers map. Must be called with Consumable mutex Locked.
-// returns true if changed, false if not
-func (c *Consumable) AllowConsumerLocked(cache *ConsumableCache, id NumericIdentity) bool {
-	consumer := c.getConsumer(id)
-	if consumer == nil {
+// AllowIngressIdentityLocked adds the given security identity to the Consumable's
+// IngressIdentities map. Must be called with Consumable mutex Locked.
+// Returns true if the identity was not present in this Consumable's
+// IngressIdentities map, and thus had to be added, false if it is already added.
+func (c *Consumable) AllowIngressIdentityLocked(cache *ConsumableCache, id NumericIdentity) bool {
+	isIdentityAllowed := c.isIdentityAllowed(id)
+	if isIdentityAllowed == false {
 		log.WithFields(logrus.Fields{
 			logfields.Identity: id,
 			"consumable":       logfields.Repr(c),
-		}).Debug("New consumer Identity for consumable")
+		}).Debug("New ingress security identity for consumable")
 		c.addToMaps(id)
-		c.Consumers[id.StringID()] = NewConsumer(id)
+		c.IngressIdentities[id] = true
 		return true
 	}
-	consumer.DeletionMark = false
+
```

Comment:
Need to `c.IngressIdentities[id] = true` here as well to mark an existing entry to not be garbage collected.

Reply:
Good catch.